include/revng/Pipeline/ReadPathsCache.h

```diff
@@ -0,0 +1,215 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <string>
+#include <tuple>
+
+#include "llvm/ADT/StringRef.h"
+#include "llvm/Support/YAMLTraits.h"
+
+#include "revng/Pipeline/Target.h"
+#include "revng/TupleTree/TupleTreePath.h"
+
+namespace pipeline {
+class Context;
+
+class ParsableLocatedTarget {
+public:
+  std::string Content;
+  std::string StepName;
+  std::string ContainerName;
+};
+
+class LocatedTarget {
```

Comment:
```suggestion
class TargetInStep {
```
?

"location" is a bit of an overloaded term.